Replace poetry with uv (#232)
fulcrumgenomics/fgpyo: CONTRIBUTING.md

## Installation for Local Development

[`uv`][uv-link] is used to manage the python development environment; [installation instructions for `uv` are here][uv-install-link].

A simple way to create an environment with the desired version of `python` and `uv` is to use a virtual environment.  E.g.:

```console

> uv venv --python 3.12

Author: ```suggestion
uv venv --python 3.12 --seed
```

source .venv/bin/activate

> # --group is required to install `mkdocs` and associated dependencies,
> # which are required for development 

Reviewer: ```suggestion
# `--group docs` is required to install `mkdocs` and associated dependencies
# `--group dev` is required to install test and build dependencies
```

uv pip install --group dev --group docs
```

> [rtd-link]:        http://fgpyo.readthedocs.org/en/stable

Author: ```suggestion
```
